cmd/xl-v1.go

```diff
@@ -236,6 +231,51 @@ func (xl xlObjects) crawlAndGetDataUsage(ctx context.Context, buckets []BucketIn
 		return err
 	}
 
+	// Collect bloom filters from all disks here...
+	bf := &bloomFilter{}
+	for _, disk := range disks {
+		hist := bloomIdx - dataUsageUpdateDirCycles
+		if bloomIdx < dataUsageUpdateDirCycles {
+			hist = 0
+		}
+
+		diskBF, err := disk.UpdateBloomFilter(ctx, hist, bloomIdx)
+		if false && intDataUpdateTracker.debug {
+			b, _ := json.MarshalIndent(diskBF, "", "  ")
+			logger.Info("Disk %v, Bloom filter: %v", disk.String(), string(b))
+		}
+
+		if err != nil || !diskBF.Complete || bf == nil {
+			logger.LogIf(ctx, err)
+			bf = nil
+			continue
+		}
+
+		if bf.BloomFilter == nil {
+			// For the first, add this as the first filter.
+			bf.BloomFilter = &bloom.BloomFilter{}
+			_, err := bf.ReadFrom(bytes.NewBuffer(diskBF.Filter))
+			if err != nil {
+				logger.LogIf(ctx, err)
+				bf = nil
+			}
+			continue
+		}
+		var tmp bloom.BloomFilter
```

Comment:
Minor comment: can simplify code here a bit (and prevent code duplication from lines 256-262 and 265-270) by always reading into tmp and either assigning tmp to bf.Bloomfilter or merging

Reply:
Awesome. Thanks!